src/main/java/org/gridsuite/study/server/service/NonEvacuatedEnergyService.java

```diff
@@ -0,0 +1,355 @@
+                generatorsLimitEntity.addGenerators(generatorsLimitByTypeEntity);
+            }
+        }
+
+        List<NonEvacuatedEnergyMonitoredBranchesEntity> monitoredBranchesEntities = new ArrayList<>();
+        if (parameters.getMonitoredBranches() != null) {
+            for (NonEvacuatedEnergyMonitoredBranches monitoredBranches : parameters.getMonitoredBranches()) {
+                NonEvacuatedEnergyMonitoredBranchesEntity monitoredBranchesEntity = new NonEvacuatedEnergyMonitoredBranchesEntity();
+                monitoredBranchesEntity.setActivated(monitoredBranches.isActivated());
+                monitoredBranchesEntity.setIstN(monitoredBranches.isIstN());
+                monitoredBranchesEntity.setLimitNameN(monitoredBranches.getLimitNameN());
+                monitoredBranchesEntity.setNCoefficient(monitoredBranches.getNCoefficient());
+                monitoredBranchesEntity.setIstNm1(monitoredBranches.isIstNm1());
+                monitoredBranchesEntity.setLimitNameNm1(monitoredBranches.getLimitNameNm1());
+                monitoredBranchesEntity.setNm1Coefficient(monitoredBranches.getNm1Coefficient());
+                monitoredBranchesEntity.setMonitoredBranches(EquipmentsContainerEmbeddable.toEmbeddableContainerEquipments(monitoredBranches.getBranches()));
+                monitoredBranchesEntities.add(monitoredBranchesEntity);
+            }
+        }
+
+        List<NonEvacuatedEnergyContingenciesEntity> contingenciesEntities = new ArrayList<>();
+        if (parameters.getContingencies() != null) {
+            for (NonEvacuatedEnergyContingencies contingencies : parameters.getContingencies()) {
+                NonEvacuatedEnergyContingenciesEntity contingenciesEntity = new NonEvacuatedEnergyContingenciesEntity();
+                contingenciesEntity.setActivated(contingencies.isActivated());
+                contingenciesEntity.setContingencies(EquipmentsContainerEmbeddable.toEmbeddableContainerEquipments(contingencies.getContingencies()));
+                contingenciesEntities.add(contingenciesEntity);
+            }
+        }
+
+        return new NonEvacuatedEnergyParametersEntity(null, stageDefinitionEntities, stagesSelectionEntities,
+            generatorsLimitEntity, monitoredBranchesEntities, contingenciesEntities);
+    }
+
+    public static NonEvacuatedEnergyParametersInfos fromEntity(NonEvacuatedEnergyParametersEntity entity) {
+        Objects.requireNonNull(entity);
+
+        List<NonEvacuatedEnergyStageDefinition> stageDefinitionParam = new ArrayList<>();
+        entity.getStagesDefinition().stream().map(stageDefinitionEntity ->
+                new NonEvacuatedEnergyStageDefinition(EquipmentsContainerEmbeddable.fromEmbeddableContainerEquipments(stageDefinitionEntity.getGenerators()),
+                    stageDefinitionEntity.getEnergySource(), stageDefinitionEntity.getPMaxPercents())
+        ).collect(Collectors.toList()).forEach(stageDefinitionParam::add);
```

Comment:
isn't is redundant to collect as a list and add it to stageDefinitionParam after that ?
Could get rid of the collector or:
List<NonEvacuatedEnergyStageDefinition> stageDefinitionParam = entity.getStagesDefinition().stream().map(stageDefinitionEntity ->
new NonEvacuatedEnergyStageDefinition(EquipmentsContainerEmbeddable.fromEmbeddableContainerEquipments(stageDefinitionEntity.getGenerators()),
stageDefinitionEntity.getEnergySource(), stageDefinitionEntity.getPMaxPercents())
).collect(Collectors.toList());
?
Same for other params

Reply:
Changed